db_spec/pg/repository/db_spec.cr

```diff
@@ -0,0 +1,59 @@
+require "../pg_spec"
+require "../../repository_spec"
+
+describe "Repository(Postgres)#db=" do
+  repo = repo(:postgresql)
+
+  context "with DB::Database" do
+    it "sets the db to the given database" do
+      old_db = repo.db
+
+      DB.open(ENV["POSTGRESQL_URL"]) do |db|
+        repo.db = db
+        new_db = repo.db
+
+        new_db.should_not be old_db
+        new_db.should be_a(::DB::Database)
+        new_db.should be db
```

Comment:
Um, this looks cumbersome. Please DRY the specs and also add actual transaction tests. These are integration tests and we have to be sure it works as intended. Assessing Crystal's equality operator is what we're aiming for 😅